Error handling for catalog not available 503

BPerlakiH opened the pull request:
Fixes: #959

### Solution: 
Additional enum case for error, and re-use existing translation key for displaying the error.

kelson42 reviewed (changes requested):
@BPerlakiH IMHO you should have a library URL.builder which should be totally disconnected from the libraryView... and the reuse it in your observer

BPerlakiH commented:
> @BPerlakiH IMHO you should have a library URL.builder which should be totally disconnected from the libraryView... and then reuse it in your observer. Havin to hardcode URL wherever you need it is not a good sign.

I think it's not that bad as it seems. Formerly we had the url hard coded in a single place in the LibraryViewModel (which is not a view). The only change is that I moved it up higher to the top of the file, so it's easier to find. That's about it, when it comes to the production code.

In tests, now I updated to use a mock url, and it works equally good with that as well. The responses are also mocks, which is they key part of those tests.
